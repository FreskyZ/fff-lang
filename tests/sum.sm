
fn main() {

    var i32 sum = 0;
    for i in 0 : 100 {
        sum += i;
    }
    writeln("1 add to 100 is " + sum.to_string());
}