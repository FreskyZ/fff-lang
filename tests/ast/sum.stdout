module tests/ast/sum.f3
  fn-def <2:1-9:1> main <2:4-2:7> () <2:8-2:9>
    block <2:11-9:1>
      var-def <4:5-4:16> mutable sum <4:9-4:11>
        literal i32 0 <4:15-4:15>
      for-stmt <5:5-7:5> iter-var i <5:9-5:9>
        range-both-expr <5:14-5:21> dotdot <5:16-5:17>
          literal i32 0 <5:14-5:14>
          literal i32 100 <5:19-5:21>
        block <5:23-7:5>
          assign-expr-stmt <6:9-6:17> += <6:13-6:14>
            path <6:9-6:11>
              segment <6:9-6:11> sum
            path <6:16-6:16>
              segment <6:16-6:16> i
      simple-expr-stmt <8:5-8:50>
        call-expr <8:5-8:49> () <8:12-8:49>
          path <8:5-8:11>
            segment <8:5-8:11> writeln
          binary-expr <8:13-8:48> + <8:32-8:32>
            literal str "1 add to 100 is " <8:13-8:30>
            call-expr <8:34-8:48> () <8:47-8:48>
              member-expr <8:34-8:46> . <8:37-8:37>
                path <8:34-8:36>
                  segment <8:34-8:36> sum
                member-name <8:38-8:46> to_string <8:38-8:46>
